### ISSUE TITLE
Suggestion: allow env argument to append variables to existing environment

### ISSUE THREAD

**Issue by Hong-Revo** [r-lib/processx#232]
- If I do `run("env")`, I get a listing of all the environment variables that are currently set.
- If I do `run("env", env=c(MYVAR="stuff"))`, I can see that only `MYVAR` is set (as well as a processx-internal variable).

It might be good to have a way to append a user-specified var to the environment, rather than replacing it. This can of course be done on the user's side, but I can see it being a generally useful feature to have.

**Comment by gaborcsardi** [comment]
I agree that this probably comes up relatively often. It is also somewhat cumbersome to write
```r
new_env <- Sys.getenv()
new_env["MYVAR"] <- "stuff"
run("env", env = new_env)
```

I am still unsure if it is worth adding a new argument for this, though. If we had an `update_vec` function, we could easily write
```r
run("env", env = update_vec(Sys.getenv(), c(MYVAR = "stuff")))
```

**Comment by Hong-Revo** [comment]
I just realised we do have an update_vec in base R: `append`

```r
x <- c(a="a", b="b")
append(x, c(z="z"))
#   a   b   z
# "a" "b" "z"
```

**Comment by gaborcsardi** [comment]
Oh, cool. One glitch is that it always appends the new variables, but it does not overwrite them if they already exist in `Sys.getenv()`. On macOS this still works, the last instance of the variable is used:

```r
❯ processx::run("bash", c("-c", "echo $TERM"))$stdout
[1] "xterm-256color\n"

❯ processx::run("bash", c("-c", "echo $TERM"), env = append(Sys.getenv(), c(TERM= "bar")))$stdout
[1] "bar\n"
```

But I am not sure if it works on all OSes.
